loama/src/components/header/HeaderBase.vue

```diff
@@ -1,47 +1,40 @@
 <template>
     <header>
         <div class="wordmark">
-            <Loama />
+            <Loama/>
             <span>Loama</span>
         </div>
         <nav>
             <slot></slot>
+            <IconButton :rightIcon="PhShareFat" class="share">Share access</IconButton>
         </nav>
         <img :src="pfpSrc" alt="User profile picture" @click="showContextMenu"/>
+        <Suspense>
+            <HeaderContextMenu class="menu" :class="{hidden: isContextMenuHidden}" @click="showContextMenu"/>
+            <template #fallback>
+                Loading context menu...
+            </template>
+        </Suspense>
     </header>
-    <aside :class="{visible: isContextMenuOpen}">
-        <div>
-            <strong>Name of Pod</strong>
-            <p>https://css12.onto-deside.ilabt.imec.be/osoc1/</p>
-        </div>
-        <label for="pod">Selected Pod</label>
-        <select>
-            <option selected>https://css12.onto-deside.ilabt.imec.be/osoc1/</option>
-            <option>https://css12.onto-deside.ilabt.imec.be/osoc1/</option>
-        </select>
-        <button @click.prevent="logout">Sign out</button>
-    </aside>
 </template>
 
 <script setup lang="ts">
 import { ref } from 'vue';
-import Loama from '../assets/loama.svg'
-import { store } from '@/store';
-import router from '@/router';
-const props = defineProps<{
+import Loama from '../../assets/loama.svg'
+import IconButton from '../IconButton.vue'
+import HeaderContextMenu from './HeaderContextMenu.vue'
+import { PhShareFat } from '@phosphor-icons/vue';
+defineProps<{
     pfpSrc: string,
-
 }>()
 
-const isContextMenuOpen = ref(false)
+const isContextMenuHidden = ref(true)
+
 
-async function logout() {
-    store.session.logout();
-    router.push('/'); 
-}
 function showContextMenu() {
```

Comment:
since the same fn is called to hide the menu, would name this more `toggleContextMenu` than `show...`